dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java

```diff
@@ -497,6 +502,39 @@ public void shouldUpdatePublisherLimitOnSubscriptionRemoval() {
     assertThat(claimedOffset).isGreaterThanOrEqualTo(0);
   }
 
+  @Test
+  public void shouldFailToCreateDispatcherIfBufferTooSmall() {
+    final ByteValue frameLength = ByteValue.ofMegabytes(1);
+    final int requiredBufferSize = (int) frameLength.toBytes() * 2 * 3;
+
+    final var builder =
+        Dispatchers.create("test")
+            .actorScheduler(actorSchedulerRule.get())
+            .maxFragmentLength(frameLength)
+            .bufferSize(frameLength);
+
+    assertThatThrownBy(() -> builder.build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Expected the buffer size to be greater than %s, but was %s. The max fragment length is set to %s.",
```

Comment:
Could you just shortly explain what the different between buffer and max fragment length is ? Is the buffer size the size of the whole buffer? And you have 3 fragments in it?

Reply:
Yes, kind of. The buffer size is the whole buffer of the Dispatcher. The max fragment size is the size of a fragment or batch which can be claimed. With the changes, the max fragment size = buffer size / 3 (= partitions) / 2 (= window size).